### PR TITLE
TMA-604: can put metrics in folders

### PR DESCRIPTION
**Pull request by kubamahnert**

### CODE REVIEW

File: lib/gooddata/models/metadata/folder.rb [hunk 1]
```diff
@@ -26,6 +26,28 @@ class << self
       def all(options = { :client => GoodData.connection, :project => GoodData.project })
         query('folder', Folder, options)
       end
+
+      def create(opts)
+        title = opts[:title]
```
Comment (yenkins) [inline lib/gooddata/models/metadata/folder.rb]: Refers to 'opts' more than self (maybe move it to another class?) (FeatureEnvy)

File: lib/gooddata/models/metadata/folder.rb [hunk 2]
```diff
@@ -26,6 +26,28 @@ class << self
       def all(options = { :client => GoodData.connection, :project => GoodData.project })
         query('folder', Folder, options)
       end
+
+      def create(opts)
```
Comment (yenkins) [inline lib/gooddata/models/metadata/folder.rb]: Has approx 6 statements (TooManyStatements)

File: lib/gooddata/models/metadata/folder.rb [hunk 3]
```diff
@@ -26,6 +26,28 @@ class << self
...
+        title = opts[:title]
+        summary = opts[:summary] || ''
+        type = opts[:type] || 'fact'
+
+        folder = {
+          'folder' => {
+            'content' => {
+              'type' => Array(type),
```
Comment (yenkins) [inline lib/gooddata/models/metadata/folder.rb]: Style/TrailingCommaInLiteral: Avoid comma after the last item of a hash.

File: lib/gooddata/models/metadata/metric.rb [hunk 4]
```diff
@@ -125,19 +130,13 @@ def execute
...
-    def expression=(value)
-      content['expression'] = value
-    end
-
     def validate
       fail 'Metric needs to have title' if title.nil?
       true
     end
 
+
```
Comment (yenkins) [inline lib/gooddata/models/metadata/metric.rb]: Style/EmptyLines: Extra blank line detected.

File: lib/gooddata/models/metadata/metric.rb [hunk 5]
```diff
@@ -125,19 +130,13 @@ def execute
...
+
```
Comment (yenkins) [inline lib/gooddata/models/metadata/metric.rb]: Style/EmptyLines: Extra blank line detected.

File: spec/integration/models/metric_spec.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
+require 'gooddata'
+
+describe GoodData::Metric do
+  before(:all) do
+    @rest_client = ConnectionHelper.create_default_connection
+    @project, * = ProjectHelper.load_full_project_implementation(@rest_client)
+    @folder = GoodData::Folder.create({project: @project, client: @rest_client, title: 'aaa', type: 'metric'})
```
Comment (yenkins) [inline spec/integration/models/metric_spec.rb]: Style/BracesAroundHashParameters: Redundant curly braces around a hash parameter.

Comment (yenkins) [inline spec/integration/models/metric_spec.rb]: Style/SpaceInsideHashLiteralBraces: Space inside { missing.

Comment (yenkins) [inline spec/integration/models/metric_spec.rb]: Style/SpaceInsideHashLiteralBraces: Space inside } missing.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/19/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/825/ : FAILURE in 55s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1093/ : SUCCESS in 1m 55s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/23/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/829/ : SUCCESS in 1m 47s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1097/ : SUCCESS in 34s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/8/ : FAILURE in 13s
- gooddata-ruby-pronto-code-review-zuul : SKIPPED
- gooddata-ruby-unit-tests-zuul : SKIPPED
- gooddata-ruby-integration-tests-zuul : SKIPPED
- gooddata-ruby-lcm-integration-tests-zuul : SKIPPED

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/10/ : SUCCESS in 16s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/832/ : SUCCESS in 2m 57s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1100/ : SUCCESS in 37s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1342/ : FAILURE in 25m 50s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/1105/ : UNSTABLE in 10m 56s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/11/ : FAILURE in 15s
- gooddata-ruby-pronto-code-review-zuul : SKIPPED
- gooddata-ruby-unit-tests-zuul : SKIPPED
- gooddata-ruby-integration-tests-zuul : SKIPPED
- gooddata-ruby-lcm-integration-tests-zuul : SKIPPED

**Comment by panjan** [comment]
ok to merge

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/15/ : SUCCESS in 14s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/846/ : SUCCESS in 2m 27s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1114/ : SUCCESS in 38s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1346/ : SUCCESS in 53m 49s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/1109/ : SUCCESS in 32m 57s

**Pull request merged**